### PR TITLE
Cgi test

### PR DESCRIPTION
**Pull request by llanahp**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,7 +69,7 @@ TEST_SCRIPT					:= $(addprefix $(SCRIPT_DIR)/, test.sh)
 # flags
 INCLUDE						:= -I$(HEAD_DIR_TEMPLATE) -I$(HEAD_DIR_CLASS) -I$(HEAD_DIR_CODE)
 CC 							:= c++
-CPPFLAGS 					:= -c -MMD -MP -Wshadow -Wall -Wextra -Werror -std=c++98 $(INCLUDE)
+CPPFLAGS 					:= -c -MMD -MP -Wshadow -Wall -Wextra -Werror -std=c++98 -g $(INCLUDE)
```
Comment (arikhativa) [inline Makefile:72]: hey why is this "g" here?

File: src/class/HTTPRequest/HTTPRequest.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+HTTPRequest::HTTPRequest(const Server virtualServer, int clientFd)
+	: _attempts(0), _virtualServer(virtualServer), _clientFd(clientFd), _response("\0")
+{
+}
+
+
+const char *HTTPRequest::SendingResponseError::what() const throw()
+{
+	return "Couldn't send response: send() failed";
```
Comment (arikhativa) [inline src/class/HTTPRequest/HTTPRequest.cpp:21]: wow this line is so cool!

File: src/class/HTTPRequest/HTTPRequest.cpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	else
+		return (UNKNOWN);
+}
+
+#include <vector>
+
+void HTTPRequest::recvRequest(void)
+{
+    int tmpRecvLen;
+    unsigned long long int maxRequestSize = 99999;
```
Comment (arikhativa) [inline src/class/HTTPRequest/HTTPRequest.cpp:68]: 99999 is a mgic number. please create a constant

File: src/class/HTTPRequest/HTTPRequest.cpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		case HTTPRequest::GET:
+			this->_response = HTTPRequestHandler::get::GET(this->_virtualServer, this->_rawRequest);
+			break;
+		case HTTPRequest::POST:
+			this->_response = HTTPRequestHandler::post::POST(this->_virtualServer, this->_rawRequest);
+			break;
+		case HTTPRequest::DELETE:
+			this->_response = HTTPRequestHandler::delete_::DELETE(this->_virtualServer, this->_rawRequest);
+			break;
+		default: //Uknown request
```
Comment (arikhativa) [inline src/class/HTTPRequest/HTTPRequest.cpp:122]: change commnet

**Review by arikhativa — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (llanahp) [reply Makefile:72]: This G is for debug

**Review by arikhativa — APPROVED** [review]
well done

**Pull request closed without merging**